sessionctx/variable/setvar_test.go

```diff
@@ -0,0 +1,155 @@
+		{"tidb_max_bytes_before_tiflash_external_join", []string{"0", "1"}},
+		{"tidb_max_bytes_before_tiflash_external_group_by", []string{"0", "1"}},
+		{"tidb_max_bytes_before_tiflash_external_sort", []string{"0", "1"}},
+		{"tidb_min_paging_size", []string{"1", "2"}},
+		{"tidb_max_paging_size", []string{"10", "20"}},
+		{"tidb_enable_cascades_planner", []string{"0", "1"}},
+		{"tidb_merge_join_concurrency", []string{"1", "2"}},
+		{"tidb_index_merge_intersection_concurrency", []string{"1", "2"}},
+		{"tidb_opt_projection_push_down", []string{"0", "1"}},
+		{"tidb_enable_vectorized_expression", []string{"0", "1"}},
+		{"tidb_opt_join_reorder_threshold", []string{"0", "1"}},
+		{"tidb_enable_index_merge", []string{"0", "1"}},
+		{"tidb_enable_extended_stats", []string{"0", "1"}},
+		{"tidb_executor_concurrency", []string{"1", "2"}},
+		{"tidb_enable_index_merge_join", []string{"0", "1"}},
+		{"tidb_enable_ordered_result_mode", []string{"0", "1"}},
+		{"tidb_enable_pseudo_for_outdated_stats", []string{"0", "1"}},
+		{"tidb_stats_load_sync_wait", []string{"0", "1"}},
+		{"tidb_cost_model_version", []string{"1", "2"}},
+		{"tidb_index_join_double_read_penalty_cost_rate", []string{"0", "1"}},
+		{"tidb_default_string_match_selectivity", []string{"0", "1"}},
+		{"tidb_enable_prepared_plan_cache", []string{"0", "1"}},
+		{"tidb_enable_non_prepared_plan_cache", []string{"0", "1"}},
+		{"tidb_plan_cache_max_plan_size", []string{"0", "1"}},
+		{"tidb_opt_range_max_size", []string{"0", "1"}},
+		{"tidb_opt_advanced_join_hint", []string{"0", "1"}},
+		{"tidb_opt_prefix_index_single_scan", []string{"0", "1"}},
+		{"tidb_store_batch_size", []string{"1", "2"}},
+		{"tidb_enable_inl_join_inner_multi_pattern", []string{"0", "1"}},
+		{"tidb_opt_enable_late_materialization", []string{"0", "1"}},
+		{"tidb_opt_ordering_index_selectivity_threshold", []string{"0", "1"}},
+		{"tidb_opt_enable_mpp_shared_cte_execution", []string{"0", "1"}},
+	}
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	for _, c := range testCases {
+		for _, setv := range c.cases {
+			tk.MustQuery(fmt.Sprintf("select /*+ set_var(%v=%v) */ @@%v", c.varName, setv, c.varName)).Check(testkit.Rows(setv))
+			r := tk.MustQuery(fmt.Sprintf("select @@%v", c.varName)).Rows()
+			tk.MustExec(fmt.Sprintf("set @@%v=default", c.varName))
+			tk.MustQuery(fmt.Sprintf("select @@%v", c.varName)).Check(r)
+		}
+	}
+}
+
+func TestSetVarStringOrEnum(t *testing.T) {
+	testCases := []struct {
+		varName string
+		cases   []string
+	}{
+		{"tidb_read_consistency", []string{"strict", "weak"}},
+		{"tidb_isolation_read_engines", []string{"tidb,tiflash", "tikv,tidb"}},
+		{"tidb_replica_read", []string{"follower", "prefer-leader"}},
+		{"tidb_partition_prune_mode", []string{"static", "dynamic"}},
+		{"mpp_version", []string{"0", "1"}},
+		{"tidb_opt_fix_control", []string{"44262:ON", "44389:ON,44823:ON"}},
+		// {"tidb_runtime_filter_type", []string{}},
```

Comment:
remove it ~

Reply:
Or I make it not updatable this time?
Leave it for you once you support other types

Reply:
No, no, let's make it updatable this time. I'm afraid I'll forget it later.